gomock/matchers.go

```diff
@@ -170,23 +170,30 @@ func (n notMatcher) String() string {
 }
 
 type regexMatcher struct {
-	regex string
+	regex      *regexp.Regexp
+	compileErr error
```

Comment:
```suggestion
	regex *regexp.Regexp
```
Rationale: `mock` is for testing and it should be okay to panic if the regular expression is invalid. The current code silently ignores invalid regular expressions, which in my opinion is not the best option during testing.